pkg/planner/core/rule/rule_predicate_simplification.go

```diff
@@ -142,12 +169,74 @@ func (*PredicateSimplification) Optimize(_ context.Context, p base.LogicalPlan,
 	return p.PredicateSimplification(opt), planChanged, nil
 }
 
+const maxInListToExpand = 1000
+
+// inListToOrList rewrites a predicate of the form
+//
+//	col IN (c1, c2, ..., cn)
+//
+// into
+//
+//	(col = c1 OR col = c2 OR ... OR col = cn).
+//
+// If the IN list has a single constant, it's simplified to col = c1.
+// Only applies if:
+//   - The left-hand side is a Column
+//   - All list items are constant expressions
+//
+// Otherwise, it returns the original expression.
+func inListToOrList(ctx base.PlanContext, expr expression.Expression) expression.Expression {
+	sf, ok := expr.(*expression.ScalarFunction)
+	if !ok || sf.FuncName.L != ast.In {
+		return expr
+	}
+
+	// First argument should be the column
+	col, ok := sf.GetArgs()[0].(*expression.Column)
```

Comment:
We could first put `sf.GetArgs()` into a variable and then read from this variable later. This way, the code would be more friendly for debugging once it's written correctly.